### PR TITLE
Fix test for Dart 2.0 ints

### PR DESCRIPTION
**Pull request by keertip**

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: typed_data
-version: 1.1.5
+version: 1.1.6
```
Comment (nex3) [inline pubspec.yaml]: Since this doesn't change any user-consumable code, can we just make it 1.1.6-dev?

File: test/typed_buffers_test.dart [hunk 2]
```diff
@@ -185,13 +185,11 @@ void testInt(int bits, buffer(int length), {String testOn}) {
...
-  0x0ffffffffffffffff,
-  0xaaaaaaaaaaaaaaaa,
-  0x8000000000000001,
-  0x8000000000000000, // 2^63
-  0x7fffffffffffffff,
+  0x1000000000000001,
+  0x7aaaaaaaaaaaaaaa,
+  0x7000000000000001,
+  0x7000000000000000,
+  0x7fffffffffffffff, //2^63
```
Comment (nex3) [inline test/typed_buffers_test.dart]: Isn't this `2^63 - 1`?

File: test/typed_buffers_test.dart [hunk 3]
```diff
@@ -267,10 +265,17 @@ void testIntBuffer(
   expect(buffer.length, equals(samples.length));
 
   // Both values are in `samples`, but equality is performed without rounding.
-  expect(buffer.contains(min - 1), isFalse);
-  expect(buffer.contains(max + 1), isFalse);
+  // for signed 64 bit ints, min and max wrap around. min-1=max and max+1=min
+  if (bits == 64){
```
Comment (nex3) [inline test/typed_buffers_test.dart]: Nit: `) {`

File: test/typed_buffers_test.dart [hunk 4]
```diff
@@ -267,10 +265,17 @@ void testIntBuffer(
   expect(buffer.length, equals(samples.length));
 
   // Both values are in `samples`, but equality is performed without rounding.
-  expect(buffer.contains(min - 1), isFalse);
-  expect(buffer.contains(max + 1), isFalse);
+  // for signed 64 bit ints, min and max wrap around. min-1=max and max+1=min
```
Comment (nex3) [inline test/typed_buffers_test.dart]: Nit: Capitalization, punctuation.

File: test/typed_buffers_test.dart [hunk 5]
```diff
@@ -185,13 +185,11 @@ void testInt(int bits, buffer(int length), {String testOn}) {
 }
 
 const List<int> intSamples = const [
-  0x10000000000000001,
-  0x10000000000000000, // 2^64
-  0x0ffffffffffffffff,
-  0xaaaaaaaaaaaaaaaa,
-  0x8000000000000001,
-  0x8000000000000000, // 2^63
```
Comment (alexmarkov) [inline test/typed_buffers_test.dart]: Consider keeping 0x8000000000000000 value if possible, as it is a corner case which is worth checking.
With the new fixed-size wrap-around integers, 0x8000000000000000 == -0x8000000000000000 == min.

File: test/typed_buffers_test.dart (hunk 2 above)
Comment (keertip) [reply test/typed_buffers_test.dart]: yup, it is.

File: pubspec.yaml (hunk 1 above)
Comment (keertip) [reply pubspec.yaml]: done.

File: test/typed_buffers_test.dart (hunk 4 above)
Comment (keertip) [reply test/typed_buffers_test.dart]: fixed.

File: test/typed_buffers_test.dart (hunk 3 above)
Comment (keertip) [reply test/typed_buffers_test.dart]: fixed, had to do without the formatter.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,7 @@
+## 1.1.6
+
+* Fix tests to work with Dart 2.0 fixed size ints.
```
Comment (nex3) [inline CHANGELOG.md]: You can also remove this changelog entry now.

File: test/typed_buffers_test.dart (hunk 5 above)
Comment (keertip) [reply test/typed_buffers_test.dart]: Works, added back.

File: test/typed_buffers_test.dart [hunk 7]
```diff
@@ -185,13 +185,12 @@ void testInt(int bits, buffer(int length), {String testOn}) {
 }
 
 const List<int> intSamples = const [
-  0x10000000000000001,
-  0x10000000000000000, // 2^64
-  0x0ffffffffffffffff,
-  0xaaaaaaaaaaaaaaaa,
-  0x8000000000000001,
```
Comment (lrhn) [inline test/typed_buffers_test.dart]: The 16-digit hex numerals (that is, all but the first two numerals) are all valid Dart 2 integer literals.
Hex literals are allowed to have values in the range -2^63 .. 2^64-1, with the upper ones being normalized to negative values. This allows you to write *bit patterns* efficiently.
So, only remove the first two, 0x10000000000000001 and 0x10000000000000000.

File: test/typed_buffers_test.dart [hunk 8]
```diff
@@ -267,10 +266,17 @@ void testIntBuffer(
...
   // Both values are in `samples`, but equality is performed without rounding.
-  expect(buffer.contains(min - 1), isFalse);
-  expect(buffer.contains(max + 1), isFalse);
+  // For signed 64 bit ints, min and max wrap around, min-1=max and max+1=min
+  if (bits == 64) {
+    // TODO(keertip): fix tests for Uint64 / In64 as now Uints are represented
+    // as signed ints.
+    expect(buffer.contains(min - 1), isTrue);
+  } else {
+    expect(buffer.contains(min - 1), isFalse);
```
Comment (lrhn) [inline test/typed_buffers_test.dart]: I see no reason these tests can't run on 64-bit values too. 
The min and max values might not really be min and max (they're 0 and -1 when represented as Int64), but the `Uint64List.contains` method will still do `int.==` to figure out if the argument is in the list, and `max + 1` is 0, and that's definitely in the list.

So, all in all, I don't see a problem with keeping the test as-is (but maybe commenting that `Uint64List` acts exactly like `Int64List` because the `int` type cannot distinguish them.

The `Uint64List` tests will not work on JavaScript anyway because `Uint64List` doesn't exist.

File: test/typed_buffers_test.dart [hunk 9]
```diff
@@ -185,13 +185,12 @@ void testInt(int bits, buffer(int length), {String testOn}) {
...
-  0x10000000000000001,
-  0x10000000000000000, // 2^64
-  0x0ffffffffffffffff,
-  0xaaaaaaaaaaaaaaaa,
-  0x8000000000000001,
+  0x1000000000000001,
+  0x7aaaaaaaaaaaaaaa,
+  0x7000000000000001,
+  0x7000000000000000,
+  0x7fffffffffffffff, // 2^63-1
```
Comment (lrhn) [inline test/typed_buffers_test.dart]: The numerals were in numerical order, so do keep that.

File: test/typed_buffers_test.dart (hunk 8 above)
Comment (keertip) [reply test/typed_buffers_test.dart]: For Uint64 
`min is 0, min-1 = -1`
`max is -1, max+1 = 0`

and for Int64
`min is -9223372036854775808, min-1 = 9223372036854775807`
`max is 9223372036854775807, max+1 = -9223372036854775808`

so these lines
```
  expect(buffer.contains(min - 1), isFalse);
  expect(buffer.contains(max + 1), isFalse);
```
fail. Since it wraps around, the values are contained in the buffer.

File: test/typed_buffers_test.dart (hunk 7 above)
Comment (keertip) [reply test/typed_buffers_test.dart]: Done.

File: test/typed_buffers_test.dart (hunk 8 above)
Comment (keertip) [reply test/typed_buffers_test.dart]: For Uint64 
`min is 0, min-1 = -1`
`max is -1, max+1 = 0`

and for Int64
`min is -9223372036854775808, min-1 = 9223372036854775807`
`max is 9223372036854775807, max+1 = -9223372036854775808`

so these lines
```
  expect(buffer.contains(min - 1), isFalse);
  expect(buffer.contains(max + 1), isFalse);
```
fail. Since it wraps around, the values are contained in the buffer.

Comment (lrhn) [reply test/typed_buffers_test.dart]: Ah, so `contains` didn't restrict the argument number to 64 bits before, but now it does. Makes sense. In that case, carry on!

File: .travis.yml [hunk 10]
```diff
@@ -1,7 +1,6 @@
 language: dart
 dart:
   - dev
-  - stable
```
Comment (kevmoo) [inline .travis.yml]: Does this really need to drop stable? I don't see BigInt being used...

Comment (keertip) [reply .travis.yml]: The except in the test returns different results on stable and dev, cause of difference in int64 behavior in the VM. Fixed it for dev16, but now it fails on stable.

Comment (kevmoo) [reply .travis.yml]: Oh...hrm...

Couldn't we modify the tests so they work on both?

My concern: if we drop testing for stable, we should likely drop it in the pubspec, too.

But it'd be nice to avoid that since – AFAICT – there is no behavior change

Comment (keertip) [reply .travis.yml]: With dev 16, min-1=max and max+1=min for int 64, so it wraps around.  So now min-1 and max+1 are contained in the buffer, whereas previously they were not. This is a corner case, but it is a change in behavior. So change the pubspec?

Comment (lrhn) [reply .travis.yml]: Sure, change the pubspec to require the next version of the SDK,

Comment (keertip) [reply .travis.yml]: @kevmoo ptal.

File: pubspec.yaml (hunk 1 above)
Comment (kevmoo) [reply pubspec.yaml]: Done?

Comment (keertip) [reply pubspec.yaml]: bumped version and updated sdk constraints. does this look ok?
